src/policy/satisfy.rs

```diff
@@ -227,112 +232,170 @@ mod tests {
         }
     }
 
+    fn execute_successful(
+        commit: Rc<CommitNode<Elements>>,
+        env: &ElementsEnv,
+        witness: Vec<Value>,
+    ) {
+        let program = commit.finalize(witness.into_iter()).expect("finalize");
+        let mut mac = BitMachine::for_program(&program);
+        assert!(mac.exec(&program, &env).is_ok());
+    }
+
     #[test]
     fn satisfy_unsatisfiable() {
         let satisfier = get_satisfier();
         let policy = Policy::Unsatisfiable;
-        assert_eq!(None, policy.satisfy(&satisfier));
+
+        assert!(policy.satisfy(&satisfier).is_none());
+
+        let mut context = Context::default();
+        let commit = policy.compile(&mut context).expect("compile");
+        let program = commit.finalize(std::iter::empty()).expect("finalize");
+        let mut mac = BitMachine::for_program(&program);
+
+        assert!(mac.exec(&program, &satisfier.env).is_err());
     }
 
     #[test]
     fn satisfy_trivial() {
         let satisfier = get_satisfier();
         let policy = Policy::Trivial;
-        assert_eq!(
-            Vec::<Value>::new(),
-            policy.satisfy(&satisfier).expect("satisfiable")
-        );
+
+        let (witness, commit) = policy.satisfy(&satisfier).expect("satisfiable");
+        assert_eq!(Vec::<Value>::new(), witness);
+
+        execute_successful(commit, &satisfier.env, witness);
     }
 
     #[test]
     fn satisfy_pk() {
         let satisfier = get_satisfier();
-        let mut keys = satisfier.keys.keys();
-        let policy = Policy::Key(*keys.next().unwrap());
+        let mut it = satisfier.keys.keys();
+        let xonly = it.next().unwrap();
+        let policy = Policy::Key(*xonly);
 
-        let witness = policy.satisfy(&satisfier).expect("sat");
+        let (witness, commit) = policy.satisfy(&satisfier).expect("satisfiable");
         assert_eq!(1, witness.len());
-        let signature = witness[0].try_to_bytes().expect("bytes");
-        assert_eq!(64, signature.len());
+
+        let sighash = satisfier.env.c_tx_env().sighash_all();
+        let message = secp256k1_zkp::Message::from(sighash);
+        let signature_bytes = witness[0].try_to_bytes().expect("to bytes");
```

Comment:
In ec2271e5eabfe3055b25652472041e15f06fb82b:

This `try_to_bytes` followed by `from_slice` thing is ok for now (especially since the existing code does it) but we really need a dedicated method to extract 32-byte arrays from `Value`s.

Reply:
Created #80